core/src/test/scala/unit/kafka/server/ListGroupsRequestTest.scala

```diff
@@ -0,0 +1,42 @@
+/*
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package kafka.server
+
+import org.apache.kafka.common.errors.UnsupportedVersionException
+import org.apache.kafka.common.message.ListGroupsRequestData
+import org.apache.kafka.common.requests.ListGroupsRequest
+import org.apache.kafka.common.requests.ListGroupsResponse
+
+import java.util.Arrays
+
+import org.junit.Assert._
+import org.junit.Test
+
+class ListGroupsRequestTest extends BaseRequestTest {
+  override val brokerCount: Int = 1
+
+  @Test(expected=classOf[UnsupportedVersionException])
+  def testListGroupsRequestOldVersionWithStates(): Unit = {
+    val request = new ListGroupsRequest.Builder(
+        new ListGroupsRequestData().setStates(Arrays.asList("Stable"))
+    ).build(3)
+
+    connectAndReceive[ListGroupsResponse](request, destination = controllerSocketServer)
+
+  }
```

Comment:
This sounds a bit heavy to verify that the serialization effectively returns an `UnsupportedVersionException`.  What about doing this in `RequestResponseTest` instead?

Reply:
Yes it was indeed pretty heavy. I've made `ListGroupsRequest.Builder.build()` throw ( as suggested in https://github.com/apache/kafka/pull/8238#discussion_r399768387 and this is now tested in `RequestResponseTest`